themes/westerndevs/scripts/author-list-json.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+hexo.extend.generator.register('author-list-json', function(locals) {
+
+  var authorList = [];
+  var authorKeys = Object.keys(locals.data.authors);
+  for (var i = 0; i < authorKeys.length; i++) {
+    var authorId = authorKeys[i];
+    var author = locals.data.authors[authorId];
```

Comment:
It looks to me like locals.data.authors is an object with keys for each author id. So you could probably do `for (var author in locals.data.authors)`